src/API/Site/Controllers/Assets/AssetSuggestionsController.php

```diff
@@ -0,0 +1,158 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Assets;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AssetSuggestionsService;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use WP_REST_Request as Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AssetSuggestionsController
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Assets
+ */
+class AssetSuggestionsController extends BaseController {
+
+	/**
+	 * Service used to populate ads suggestions data.
+	 *
+	 * @var AssetSuggestionsService
+	 */
+	protected $asset_suggestion_service;
+
+	/**
+	 * AssetSuggestionsController constructor.
+	 *
+	 * @param RESTServer              $server
+	 * @param AssetSuggestionsService $asset_suggestion
+	 */
+	public function __construct( RESTServer $server, AssetSuggestionsService $asset_suggestion ) {
+		parent::__construct( $server );
+		$this->asset_suggestion_service = $asset_suggestion;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'assets/final-url/suggestions',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_final_urls_suggestions_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_collection_params(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params(): array {
+		return [
+			'search'   => [
+				'description'       => __( 'Search for post title or term name', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'default'           => '',
+				'validate_callback' => 'rest_validate_request_arg',
```

Comment:
Maybe we can do some sanitisation as the original search param does.

https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-controller.php#L364